test/grpc/route_guide/record-route.cc

```diff
@@ -0,0 +1,67 @@
+#include <string>
+
+#include "eventuals/finally.h"
+#include "eventuals/foreach.h"
+#include "eventuals/head.h"
+#include "eventuals/iterate.h"
+#include "eventuals/let.h"
+#include "eventuals/promisify.h"
+#include "eventuals/range.h"
+#include "eventuals/then.h"
+#include "gtest/gtest.h"
+#include "test/grpc/route_guide/route-guide-test.h"
+
+using eventuals::grpc::Stream;
+using grpc::Status;
+
+using eventuals::Finally;
+using eventuals::Foreach;
+using eventuals::operator*;
+using eventuals::Head;
+using eventuals::Iterate;
+using eventuals::Let;
+using eventuals::Range;
+using eventuals::Then;
+
+using routeguide::Point;
+
+TEST_F(RouteGuideTest, RecordRouteTest) {
+  const int points = 10;
+  std::vector<Point> data(points);
+  CHECK_LE(points, feature_list.size());
+
+  for (size_t i = 0; i < points; ++i) {
+    data[i] = feature_list[i].location();
+  }
+
+  auto e = [&]() {
+    return client().RecordRoute()
+        >> Then(Let([&data](auto& call) mutable {
+             return Foreach(
+                        Iterate(std::move(data)),
+                        ([&call](auto&& request) {
+                          return call.Writer().Write(request);
+                        }))
+                 >> call.WritesDone()
+                 >> call.Reader().Read()
+                 >> Head()
+                 >> Finally(Let([&call](auto& response) {
+                      return call.Finish()
+                          >> Then([&](Status&& status) {
```

Comment:
Again, let's spell fully, here and everywhere else please:
```suggestion
                          >> Then([&](::grpc::Status&& status) {
```